src/lib/queryFromFilters.js

```diff
@@ -104,6 +104,9 @@ function addConstraint(query, filter) {
     case 'keyLte':
       addQueryConstraintFromObject(query, filter, 'lessThanOrEqualTo');
       break;
+    case 'dneOrNull':
+      query.equalTo(filter.get('field'), null);
```

Comment:
I don't think this query will return null values.

Reply:
True, it will also include records which does not have filter field even set, that's why the name of filter `does not exits or null`

Reply:
That sounds good

> I think only one filter is allowed on a field at a time.

You can set composable to true if you to do multiple filters on a field. Similar to how less than does it. (Perhaps we should document these)

Reply:
The difference is the MongoDB performance. The current query is an explicit "does not exist or is null" conditional, without `$and`. This distinction is also relevant for indexing.

Reply:
I mean document `composable` and `comparable` it usually takes a while to figure out what they mean.

Reply:
`field` is used when `comparable` is true. We can document that too

Reply:
Thanks @dplewis for `composable` idea. @mtrezza I tried that but it seems like `exists` is not composable. 

I was selecting first `does not exists or null` & then `exists` for a field, but it seems to send query request with only `where={age:{$exists: true}}.

And other thing is that, if `exists` is not composable then we should even not be able to select that.

Reply:
> this is what I tried,
> 1. select does not exists or null --> query.equalTo(field, null)
> 2. select exists for a field, --> query.exists(field)
> but it seems to send query request with only `where={age:{$exists: true}}.

It sends `where={age:{$exists: true}}` if you set a combined filter of (1) and (2)?

Reply:
yes. 

I think that's strange as, for let's say 2 filters `less than` & `exists`, we do something like
```js
query.lessThan(field, compareTo); // compareTo is value entered by user to which s(he) wants to compare
query.exists(field);
```
This sends a query like `where={age: {$lt: 4, $exists: true}}`

Then, why not this
```js
query.equalTo(field, null);
query.exists(field);
```
sends a query like `where={age: {$eq: null, $exists: true}}`

Reply:
@dplewis when you mentioned `composable`, on what level did you mean that? On a MongoDB level?

Reply:
I don't think it's a limitation with MongoDB rather it's with JS Sdk.(just an opinion) 

As when I explicitly(using cURL) make a request with `where={age: {$exists: true, $eq: null }}`, it does return records only with `null` field value.

But, when I'm trying to couple `$exists` & `$eq` query, it only sends request with `where={age: {$exists: true}}`.
I'm trying this way to construct the query
```js
query.equalTo(field, null);
query.exists(field);
```
Maybe I'm missing some point with js sdk. Would really appreciate any help with creating a query with `where={age: {$exists: true, $eq: null }}`.

Reply:
I created a new issue for this https://github.com/parse-community/Parse-SDK-JS/issues/1372.